ase/src/main/java/com/ase/ase/AseCovidDashboardApplication.java

```diff
@@ -26,29 +24,18 @@ public static void main(String[] args) {
   }
 
   @Bean
-  public CommandLineRunner demo(
-      DownloadService service,
-      DownloadPopulationData downloadPopulationData,
-      OverviewRepository repository,
+  public CommandLineRunner start(
+      DownloadService downloadService,
+      DownloadScheduleService scheduleService,
+      DownloadPopulationData downloadPopulationService,
       MessagingService messagingService
   ) {
     messagingService.init("new-data");
-    repeatPushingDataForDemo(messagingService);
+    scheduleService.setTimerForDownloadingNewData(messagingService, downloadService);
```

Comment:
A service object like messagingService should not be passed to the ScheduleService via parameters. It is better to use an AutoWired variable of type MessagingService in the ScheduleService.